### PR TITLE
Add and refine document

### PR DESCRIPTION
**Pull request by haofanwang**
This PR makes a contribution to #2510 

1. Refine tutorial.
2. Remove unnecessary dependencies. 
3. Add document.

### CODE REVIEW

**Comment by haofanwang** [comment]
Reformat done, ready to merge.

File: test/federated/test_federated_client.py [hunk 1]
```diff
@@ -11,43 +11,51 @@
...
-
+    # Create a dataset
     dataset = "my_dataset"
     key = "string_dataset"
+    # add new dataset
     fed_client.add_dataset(dataset, key)
 
     assert key in fed_client.datasets
-
+    # remove new dataset
```
Comment (gmuraru) [inline test/federated/test_federated_client.py]: Could you use 'R' in place of 'r' such that it follows the same pattern as the above comment ("Create a dataset")

File: test/federated/test_federated_client.py [hunk 2]
```diff
@@ -11,43 +11,51 @@
...
 
 
 def test_remove_dataset():
+    # Create a client to execute federated learning
     fed_client = FederatedClient()
-
+    # Create a dataset
     dataset = "my_dataset"
     key = "string_dataset"
+    # add new dataset
```
Comment (gmuraru) [inline test/federated/test_federated_client.py]: Could you use 'A' in place of 'a' such that it follows the same pattern as the above comment ("Create a dataset")

File: test/federated/test_federated_client.py [hunk 3]
```diff
@@ -11,43 +11,51 @@
...
+    # Create a dataset
     dataset = "my_dataset"
-    fed_client.add_dataset(dataset, "string_dataset")
+    key = "string_dataset"
+    # add new dataset
+    fed_client.add_dataset(dataset, key)
 
     assert "string_dataset" in fed_client.datasets
-
+    # Raise error if the key is already exists
```
Comment (gmuraru) [inline test/federated/test_federated_client.py]: ```Raise an error```

File: test/federated/test_federated_client.py [hunk 4]
```diff
@@ -11,43 +11,51 @@
...
 
 def test_add_dataset_with_duplicate_key():
+    # Create a client to execute federated learning
     fed_client = FederatedClient()
-
+    # Create a dataset
     dataset = "my_dataset"
-    fed_client.add_dataset(dataset, "string_dataset")
+    key = "string_dataset"
+    # add new dataset
```
Comment (gmuraru) [inline test/federated/test_federated_client.py]: Could you use 'A' in place of 'a' such that it is the same as above.

File: test/federated/test_federated_client.py [hunk 5]
```diff
@@ -11,43 +11,51 @@
...
 
 def test_add_dataset():
+    # Create a client to execute federated learning
     fed_client = FederatedClient()
-
+    # Create a dataset
     dataset = "my_dataset"
-    fed_client.add_dataset(dataset, "string_dataset")
+    key = "string_dataset"
+    # add new dataset
```
Comment (gmuraru) [inline test/federated/test_federated_client.py]: Could you use 'A' in place of 'a' such that it is the same as above (with big "C")?

File: syft/workers/virtual.py [hunk 6]
```diff
@@ -6,12 +6,15 @@
 
 class VirtualWorker(BaseWorker, FederatedClient):
     def _send_msg(self, message: bin, location: BaseWorker) -> bin:
+        """send message to worker location"""
         if self.message_pending_time > 0:
             if self.verbose:
                 print(f"pending time of {self.message_pending_time} seconds to send message...")
+            # pending
```
Comment (gmuraru) [inline syft/workers/virtual.py]: Q: Maybe remove this comment - I think the attribute name is self explanatory.

**Comment by gmuraru** [comment]
LGTM

**Pull request merged**